src/core/library.c

```diff
@@ -2556,3 +2570,27 @@ MsQuicExecutionPoll(
 }
 
 #endif
+
+// Allows external services to set the retry key configuration for stateless retry tokens at runtime.
+// The BaseRetrySecret length must match the AEAD algorithm requirements.
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicLibrarySetRetryKeyConfig(
+    _In_ const QUIC_RETRY_KEY_CONFIG* Config
+    )
+{
+    if (Config == NULL || Config->BaseRetrySecret == NULL) {
```

Comment:
SAL already requires `Config` to be non-NULL. Just DBG assert that instead of checking.